lib/puppet/type/iis_virtual_directory.rb

```diff
@@ -46,7 +47,22 @@
     end
   end
 
+  newproperty(:user_name, :parent => PuppetX::PuppetLabs::IIS::Property::String) do
+    desc "Specifies the identity that should be impersonated when accessing the physical path."
+  end
+
+  newproperty(:password, :parent => PuppetX::PuppetLabs::IIS::Property::String) do
```

Comment:
One thing that we're going to want to do here is use the `Sensitive` type so passwords don't show up in logs.  Sensitive was added in Puppet 4.6, so this module should support it.

Given there are other passwords on other types, I will file another ticket for doing a sweep for sensitive properties that should be using `Sensitive`.